torchvision/datasets/usps.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import print_function
+import torch.utils.data as data
+from PIL import Image
+import os
+import tqdm
+import os.path
+import numpy as np
+from .utils import download_url
+
+
+class USPS(data.Dataset):
+    """`USPS <https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/multiclass.html#usps>`_ Dataset.
+    The data-format is : [label [index:value ]*256 \n] * num_lines, where ``label`` lies in ``[1, 10]``.
+    The value for each pixel lies in ``[-1, 1]``. Here we transform the ``label`` into ``[0, 9]``
+    and make pixel values in ``[0, 255]``.
+
+    Args:
+        root (string): Root directory of dataset to store``USPS`` data files.
+        split (string): One of {'train', 'test'}.
+            Accordingly dataset is selected.
+        transform (callable, optional): A function/transform that  takes in an PIL image
+            and returns a transformed version. E.g, ``transforms.RandomCrop``
+        target_transform (callable, optional): A function/transform that takes in the
+            target and transforms it.
+        download (bool, optional): If true, downloads the dataset from the internet and
+            puts it in root directory. If dataset is already downloaded, it is not
+            downloaded again.
+
+    """
+    split_list = {
+        'train': [
+            "https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/multiclass/usps.bz2",
+            "usps.bz2", 7291
+        ],
+        'test': [
+            "https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/multiclass/usps.t.bz2",
+            "usps.t.bz2", 2007
+        ],
+    }
+
+    def __init__(self,
+                 root,
+                 split='train',
+                 transform=None,
+                 target_transform=None,
+                 download=False):
+        self.root = os.path.expanduser(root)
+        self.transform = transform
+        self.target_transform = target_transform
+        self.split = split  # training set or test set
+
+        if self.split not in self.split_list:
+            raise ValueError('Wrong split entered! Please use split="train" '
+                             'or split="test"')
+
+        self.url = self.split_list[split][0]
+        self.filename = self.split_list[split][1]
+        self.total_images = self.split_list[split][2]
+
+        full_path = os.path.join(self.root, self.filename)
+
+        if download and not os.path.exists(full_path):
+            self.download()
+
+        import bz2
+        fp = bz2.open(full_path)
+
+        datas = []
+        labels = []
+        for line in tqdm.tqdm(
```

Comment:
Is tqdm part of torchvision?

Reply:
I will switch to ``from torch.utils.model_zoo import tqdm``.